src/tdsops.f90

```diff
@@ -0,0 +1,682 @@
+module m_tdsops
+   use m_common, only: dp, pi
+
+   implicit none
+
+   type :: tdsops_t
+      !! Tridiagonal Solver Operators class.
+      !! Operator arrays are preprocessed in this class based on the arguments
+      !! provided. dist_fw and dist_bw are used in the first phase of the
+      !! distributed tridiagonal solver algorithm. dist_sa and dist_sc are used
+      !! in the final substutition phase. See the kernels_dist.f90 files in the
+      !! relevant backend folders.
+      !! coeff arrays define the specific rules of building the RHS
+      !! corresponding to the tridiagonal system to be solved, and used only in
+      !! the first phase of the distributed algorithm when building the RHS.
+      !! If a boundary condition is defined then coeffs_t and coeffs_b differ
+      !! from coeffs array and define the RHS rule for the first and last 4
+      !! entries in the tridiagonal system (n_halo = 4).
+      !!
+      !! This class does not know about the current rank or its relative
+      !! location among other ranks. All the operator arrays here are used when
+      !! executing a distributed tridiagonal solver phase one or two.
+      real(dp), allocatable, dimension(:) :: dist_fw, dist_bw, dist_sa, dist_sc
+      real(dp), allocatable :: coeffs(:), coeffs_t(:, :), coeffs_b(:, :)
+      integer :: n, n_halo
```

Comment:
Do we need default values for `n`, `n_halo`, `alpha` and `delta` ?

Reply:
putting n_halo a side (I'll start a different thread for that), we always want user to input an 'n', and should compain if they don't supply one. Its essential and I don't see any natural default number for 'n'. Alpha is actually an output, and it depends on the specifics of the operation user wants to execute. If the operation someone ones to execute is not yet available, the idea is that user adds something here in this class to make it available, and then calls the tdsops with correct arguments.
Again for delta, it is something we want user to input, I don't think there is a natural default for such a value and assuming something is just doesn't make sense to me.